src/components/DataBox.vue

```diff
@@ -1,479 +1,416 @@
-    document.removeEventListener('click', this.handleClickOutside);
-
-    // Nettoyage du ResizeObserver
-    if (this.resizeObserver) {
-      this.resizeObserver.disconnect(); // Arrête l'observation
-    }
+  date: {
+    type: String,
+    required: true,
   },
-  methods: {
-    changeDateFormat,
-    emitOpenModal() {
-      this.$emit('open-modal');
-    },
-    updateHeight() {
-      const chartContainer = this.$refs.chartContainer;
-      if (chartContainer) {
-        // Utilisation de requestAnimationFrame pour éviter la boucle d'updates
-        requestAnimationFrame(() => {
-          const parentHeight = chartContainer.parentElement.offsetHeight;
-          const parentWidth = chartContainer.parentElement.offsetWidth;
-          // Ajuster la hauteur du conteneur de manière appropriée
-          const newHeight = Math.min(parentHeight, parentWidth * 0.6); // Par exemple, 60% de la largeur
-          this.$el.style.setProperty('--chart-height', `${newHeight}px`);
-        });
-      }
-    },
-    toggleView(viewType) {
-      this.showGraph = viewType === 'graphique';
-      this.updateHeight();
-    },
-    toggleDropdown() {
-      this.isDropdownOpen = !this.isDropdownOpen;
-    },
-    handleClickOutside(event) {
-      if (this.$refs.dropdownContainer && !this.$refs.dropdownContainer.contains(event.target)) {
-        this.isDropdownOpen = false;
-      }
-    },
-    transfertSourceOption(selectedOption) {
-      this.$emit('select-source-api', selectedOption);
-    },
-    handleChartSelected(type) {
-      this.localShowGraph = type === 'graphique';
-    },
-    performAction(action) {
-      if (typeof this[action.action] === 'function') {
-        this[action.action](); // Exécute la méthode correspondante
-      } else {
-        console.warn(`Action ${action.action} non définie`);
-      }
-      this.isDropdownOpen = false;
-    },
-    actionBtn1() {
-      // Ici, logique pour la première action (par exemple, capture d'écran)
-      alert('Action 1 effectuée !');
-    },
-    actionBtn2() {
-      // Ici, logique pour la deuxième action (par exemple, télécharger CSV)
-      alert('Action 2 effectuée !');
-    },
+  defaultSource: {
+    type: String,
+    default: null,
   },
+  trend: {
+    type: String,
+    default: null,
+  },
+  segmentedControl: {
+    type: [Boolean, String],
+    default: true,
+  },
+  fullscreen: {
+    type: [Boolean, String],
+    default: false,
+  },
+  screenshot: {
+    type: [Boolean, String],
+    default: false,
+  },
+  download: {
+    type: [Boolean, String],
+    default: false,
+  },
+  modalTitle: {
+    type: String,
+    default: '',
+  },
+  modalContent: {
+    type: String,
+    default: '',
+  },
+});
+
+const chartSources = ref([]);
+const tableSources = ref([]);
+
+chartSources.value = [...document.querySelectorAll(`[databox-id="${props.id}"][databox-type="chart"]`)].map((el) => el.getAttribute('databox-source') || 'default');
```

Comment:
Tous les libellés affichés doivent être en français : 
```suggestion
chartSources.value = [...document.querySelectorAll(`[databox-id="${props.id}"][databox-type="chart"]`)].map((el) => el.getAttribute('databox-source') || 'défaut');
```

Reply:
on peut laisser le fallback tel quel et corriger l'exemple pour toujours avoir un `databox-source` et éviter de se retrouver avec ce résultat : 
![Capture d’écran 2025-01-16 à 10 29 16](https://github.com/user-attachments/assets/0a46b386-5efd-4abc-9dfb-06f676543f8a)